substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/DirEntry.java

```diff
@@ -37,11 +37,9 @@
  */
 public class DirEntry {
```

Comment:
Do we really still need `DirEntry` or should we simplify this and just use `Path`?

Reply:
Yes, I guess so. Although I like the fact that using this class clearly demarcates the Path as a directory and not, say, a file.

n.b. It's certainly not costing much to wrap the path. The Quarkus app that manifested the performance problem has 500 dirs with full debug info. The number is much less if `TrackNodeSourcePosition` or `SourceLevelDebug` is omitted.

Reply:
Sure, if you prefer a clear separation between dir and file, keep the wrapper. If you want to optimize SLOC, you can even use a record ;)